.kokoro/build-with-run.sh

```diff
@@ -37,12 +37,12 @@ gcloud config set project $GOOGLE_CLOUD_PROJECT
 
 # Version is in the format <PR#>-<GIT COMMIT SHA>.
 # Ensures PR-based triggers of the same branch don't collide if Kokoro attempts
-# to run them concurrently.
+# to run them concurrently. Defaults to 'latest'.
 export SAMPLE_VERSION="${KOKORO_GIT_COMMIT:-latest}"
 export SAMPLE_NAME="$(basename $(pwd))"
 
-# Builds not triggered by a PR will fall back to the commit hash then "latest".
-SUFFIX=${KOKORO_BUILD_ID}
+# Cloud Run has a max service name length, $KOKORO_BUILD_ID is too long to guarantee no conflict deploys.
+SUFFIX=$(cat /dev/urandom | LC_CTYPE=C tr -dc 'a-z0-9' | head -c 15)
```

Comment:
Can we use `uuid=$(uuidgen)`?   But either way is fine.

Reply:
Cloud Run has a max name length of 63 characters. This is the fix we are implementing to prevent that since commit sha and build id are usually too long.